SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LayeredLBStrategy.cpp

```diff
@@ -104,10 +98,16 @@ bool ULayeredLBStrategy::ShouldHaveAuthority(const AActor& Actor) const
 		return false;
 	}
 
-	const FName& LayerName = GetLayerNameForActor(Actor);
+	const AActor* EffectiveActor = &Actor;
+	while (EffectiveActor->GetOwner() != nullptr)
+	{
+		EffectiveActor = EffectiveActor->GetOwner();
+	}
+
+	const FName& LayerName = GetLayerNameForActor(*EffectiveActor);
 	if (!LayerNameToLBStrategy.Contains(LayerName))
 	{
-		UE_LOG(LogLayeredLBStrategy, Error, TEXT("LayeredLBStrategy doesn't have a LBStrategy for Actor %s which is in Layer %s."), *AActor::GetDebugName(&Actor), *LayerName.ToString());
+		UE_LOG(LogLayeredLBStrategy, Error, TEXT("LayeredLBStrategy doesn't have a LBStrategy for Actor %s which is in Layer %s."), *AActor::GetDebugName(EffectiveActor), *LayerName.ToString());
 		return false;
```

Comment:
This is the error case I realise, but I think the default return should be true. If it's false, ShouldHaveAuthority will mean the Actor attempts to migrates every possible tick?

Reply:
Currently we're consistent about returning false from all our LB strategies in case of error here, but I agree it might be better to return true. If we're going to make that change though, I'd rather do it in one pass separately.